org/mozilla/jss/util/NativeProxy.java

```diff
@@ -176,10 +204,16 @@ public long getRegistryIndex() {
      * is thrown.
      */
     public synchronized static void assertRegistryEmpty() {
```

Comment:
Any idea who's calling this method? The method doesn't throw exception or return anything, so this is not really an assertion.

Reply:
So it doesn't throw an assertion any more but it does log output. In master, [you're the one](https://github.com/dogtagpki/jss/commit/e0911cf5f30afba57afef522da8b7a63a46e0137) who last touched it to use `logger` instead of `Debug.trace(...)`.

Nothing that [isn't commented out](https://github.com/dogtagpki/jss/blob/master/org/mozilla/jss/tests/SelfTest.java#L221) (which isn't run through JSSTests yet) calls it formally. I've been thinking about adding `assertRegistryEmpty(...)` to the test suite for SSLEngine stuff though. I've added it back in and called it a few times in the TestBufferPRFD; that's where the enhancements for Backtracing NativeProxy creation came from.

There's only [four](https://www.google.com/search?client=firefox-b-1-d&q=assertRegistryEmpty) results for it on Google.

I don't think it'll be missed if we do drop it (or better -- quit registering objects outside of "Debug" mode so we only take its small performance hit there), but FWIW, it is kinda useful for catching leaks.

Reply:
Yeah, if we want to keep it we probably can attach it into the server shutdown code such that we get a report whether the resources are cleaned up properly. We can take care of this another time.